Fatih puyll requesttt

fatihoezkan opened the pull request.

fatihoezkan committed:
Add PPO and TD3 agents with training and loading functionalities

- Introduced new PPO and TD3 agents in the training module.
- Updated the setup_environment function to accept agent type and adjust environment parameters accordingly.
- Implemented training functions for PPO and TD3, including model saving.
- Added loading functions for trained PPO and TD3 models in utils.py.
- Enhanced result merging functionality to include PPO and TD3 agent results.
- Added binary model files for PPO and TD3 agents.

fatihoezkan committed:
Add PPO and TD3 model predictions to inference function

fatihoezkan committed:
Add hyperparameter optimization for PPO, A2C, SAC, and TD3 using Optuna

fatihoezkan committed:
n

fatihoezkan committed:
Add best hyperparameters for PPO model configuration

fatihoezkan committed:
Refactor hyperparameter optimization: remove old files and add new implementation for PPO, A2C, SAC, and TD3 models

fatihoezkan committed:
Refactor hyperparameter optimization: streamline code structure and improve readability

fatihoezkan committed:
Add JSON export for best trial hyperparameters and create SAC trials log

fatihoezkan committed:
Add best hyperparameters and trial logs for SAC and TD3 models; remove obsolete sentiment ensemble file

fatihoezkan committed:
Implement sentiment-aware trading strategy with risk-adjusted rewards and ensemble model predictions

- Created a sentiment-aware environment that incorporates risk scores into the trading strategy.
- Adjusted reward calculations based on risk levels to enhance model performance.
- Developed a dynamic weighting strategy for model predictions based on Sharpe ratios and current risk levels.
- Combined actions from multiple models using risk-adjusted weights and normalized outputs.
- Added comprehensive error handling and data integrity checks throughout the process.
- Generated performance metrics and visualizations to compare individual models and the ensemble approach.
- Saved results and plots for further analysis and review.

fatihoezkan committed:
Add ensemble agent implementation with turbulence calculation and model training

fatihoezkan committed:
Add ensemble results and risk-return analysis visualization

- Created a new CSV file `result_with_ensemble.csv` containing the account value over time for the ensemble model.
- Added a new PNG file `risk_return_analysis.png` for visual representation of risk-return analysis.

fatihoezkan committed:
Implement code changes to enhance functionality and improve performance

fatihoezkan committed:
Add turbulence analysis script for hourly price data

- Implemented turbulence calculation and bin assignment using hourly price data.
- Added functions to analyze returns by turbulence bin and visualize results.
- Created output directory and saved turbulence scores, updated price data, performance analysis, and turbulence bins to CSV files.
- Included plotting functionality to visualize average returns by turbulence bin.

fatihoezkan committed:
Refactor sentiment model predictions and enhance risk-adjusted metrics calculation

fatihoezkan committed:
Remove MetaAgent class and associated weighting functions from meta_agent.py

cassandra-soong commented on ensemble_agent.py:
why is this file located in the root directory? if you still need this, can you move it into the finrl folder?

fatihoezkan replied:
i deleted it after i pull

cassandra-soong commented on finrl/Notebook_fatih/fatih_trained_4agent.ipynb:
the main branch already has a "Notebook" folder. Can you move this file into the "Notebook" folder? 
and maybe rename it to something like 'train_all_agents.ipynb'

cassandra-soong commented on finrl/Turbulance/trade_turbulence_scores.csv:
was this turbulence score used in the agents? if so, where?

fatihoezkan replied:
no not yet

cassandra-soong commented on finrl/ensemble/custom_ensemble_model_sentiment.py:
is the ensemble folder still necessary since ensemble agents are not used? or is it just kept as a reference?

fatihoezkan replied:
i just want to keep them, because i will explain them in the report.

cassandra-soong replied:
ok

cassandra-soong commented on finrl/training.py:
```diff
@@ -7,14 +7,22 @@
         
     """ 
+    AGENT_ENV_OVERRIDES = {
+        "a2c": {"reward_scaling": 1e-4, "hmax": 500},
```
can I ask why was the 'reward_scaling' for a2c changed to 1e-4 compared to the original 1e-10?

fatihoezkan replied:
it was performing better on my local, more reward for ppo and a2c because others are more agressive so i keep low their rewards

cassandra-soong commented on finrl/utils.py:
can you also update the finrl/README.md file to include PPO, TD3 and to describe the new folders (Turbulance, HyperParameter... ) ?

fatihoezkan commented on finrl/inference.py:
```diff
@@ -27,6 +27,14 @@ def get_inference():
+
+        #load the pre-trained PPO model
+        trained_ppo = load_trained_ppo()
+        print("Loaded trained PPO model.")
```
i did actually dont know why i it didnt work.

cassandra-soong replied:
if you only did a git pull in the main branch, then it only updates your local main branch, but this new change still has not been added to your fatih_clean branch. To get your fatih_clean branch updated as well, you still have to do in your branch:
git fetch origin main
git merge origin/main

But this will be more complicated because you would have to resolve a lot of conflicts. 

So like I suggested, the easiest way for you is to create a new copy of the main branch as a new branch, manually add your files there, then create a new pull request from that new branch. We will delete this pull request.

fatihoezkan committed:
Add trained models for various reinforcement learning agents

- Added A2C agent model (agent_a2c.zip)
- Added PPO agent model (agent_ppo.zip)
- Added SAC agent model (agent_sac.zip)
- Added TD3 agent model (agent_td3.zip)

also hyper opt done with 10 trial, trained models with new learning rates
